File: php/testcode/benchmark_test_00596.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00596(BenchmarkRequest $req): BenchmarkResponse {
    $key = random_bytes(32);
    $data = $req->post('data');
    $mac = hash_hmac('sha256', $data, $key);
    return BenchmarkResponse::json(['mac' => $mac, 'key_id' => bin2hex(substr($key, 0, 4))]);
}

File: rust/testcode/benchmark_test_00596.rs
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let _user = req.param("user");

    let ts = get_timestamp_ns();
    let token = format!("{:016x}", ts);

    super::shared::BenchmarkResponse::ok(&format!("Token: {}", token))
}

fn get_timestamp_ns() -> u64 {
    1_700_000_000_123_456_789
}
